skimage/graph/_graph.py

```diff
@@ -63,7 +63,7 @@ def pixel_graph(image, *, mask=None, edge_function=None, connectivity=1, spacing
 
     Returns
     -------
-    graph : scipy.sparse.csr_matrix
+    graph : scipy.sparse.csr_array
```

Comment:
Will this still accept people passing in `csr_matrix`?

Reply:
Yes, I think this is a breaking deprecation, but one we'll have to swallow if we want to finalize transitioning to sparse arrays.

Reply:
We could raise a warning, if you want. And of course a note in the release notes.

Reply:
Discussed a bit more with @lagru; it is true that returning an array can silently break further user code, so I think we'll have to add a return_type flag, and we can raise if that flag is not set, so users will be forced to transition. In skimage2 we can make the default arrays. @lagru said he could work on a PR now.